### PR TITLE
swap order of optimizer.step() and scheduler.step()

### PR DESCRIPTION
**Pull request by adai183**
The current code results in the following warning:

```
UserWarning: Detected call of `lr_scheduler.step()` before `optimizer.step()`. In PyTorch 1.1.0 and later, you should call them in the opposite order: `optimizer.step()` before `lr_scheduler.step()`.  Failure to do this will result in PyTorch skipping the first value of the learning rate schedule.See more details at https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate
  "https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate", UserWarning)
```

### CODE REVIEW

**Comment by thomwolf** [comment]
indeed, thanks @adai183

**Comment by adai183** [comment]
🤗
